R/download.R

```diff
@@ -12,9 +12,11 @@
 #' @param bugzilla_site URL to specific bugzilla site
 #' @param start_timestamp when to start bug retrieval (ex. 2023-01-01T00:14:57Z)
 #' @param save_folder_path the full *folder* path where the bugzilla issues will be stored
+#' @param limit the number of issues saved in each page file. Some bugzilla sites have limits set on how many bugs
+#' can be retrieved in one GET request, in which case, the limit should be set accordingly.
 #' @seealso \code{\link{download_bugzilla_comments_from_rest_api}} and \code{\link{parse_bugzilla_crawler_issue}}
 #' @export
-download_bugzilla_issues_from_rest_api <- function(bugzilla_site, start_timestamp, save_folder_path){
+download_bugzilla_issues_from_rest_api <- function(bugzilla_site, start_timestamp, save_folder_path, limit=500){
```

Comment:
I added a new parameter "limit" with the default value 500, since that's what Perceval's Bugzilla REST API backend used, as we discussed here https://github.com/sailuh/kaiaulu/pull/177#discussion_r1162589675.
The reason I added the limit parameter is that some Bugzilla sites limit the number of bugs that can be retrieved in a single GET request. For instance, Red Hat Bugzilla limits the bugs retrieved to 20 by default. If you use a limit > 20, it will skip over bugs. And I didn't see a way to retrieve the default limit for a Bugzilla site manually. So I made it so a user should be able to set this value.

Reply:
> Red Hat Bugzilla limits the bugs retrieved to 20 by default. 

What do you mean by default here? If I user type limit = 500 on function call, Red Hat will accept the function call but download 20 at a time? And how did you find it was 20 by default? 

I would include this comment and the answer to my question on the function documentation body too. It sounds helpful.

Reply:
Hi Carlos, yes that's correct. The limit of 20 - for unauthenticated users specifically - is mentioned [here](https://communityblog.fedoraproject.org/changes-to-bugzilla-queries/#:~:text=The%20value%20of%20limit%20can,0%20as%20a%20limit%20parameter.)

So for a single GET request from Red Hat Bugzilla, only 20 bugs will be retrieved at a time. If a user was to specify a limit of 500, the first GET request would retrieve bugs 0-20, and the second would retrieve bugs 500-520, and so on, since I am using the limit to adjust the "offset" parameter, which determines which bugs to retrieve for each GET request.

Reply:
Interesting. One more question here, and the other I am moving to the specific part of the code.

How did you find this page? Via google? Or did their Bugzilla tracker had at the bottom of the page some link that led you there? If Bugzilla has a standard way to find these pages, it would be very helpful to who is using this function since you already went through this.